### ISSUE TITLE
Memory problem: 406MB RAM used on Firefox 35

### ISSUE THREAD

**Issue by arrarrlol** [EFForg/privacybadgerfirefox#285]
http://i.imgur.com/RQa5bXc.png

Running the latest Firefox 35 on Windows 7 64-bit.Over a period of 1 hour, I browsed about 100 Amazon,eBay,Google and some random pages and the RAM went up a lot.About 406MB were being used by Privacy Badger.I'm not a programmer but if you need more info, I can try to provide it from the file.The json.gz file contains some private links so I can't upload it.

**Comment by cooperq** [comment]
Thanks for bringing this to our attention. It sounds like there may be a memory leak going on. I will take a look at the code and try to fix it.

**Comment by erikg** [comment]
I'm seeing heavy memory usage with ff36.0.4 (and several previous versions of ff) on mac osX10.10, as well

about:memory gives me
  ├──504.21 MB (48.13%) -- jid1-MnnxcxisBPnSXQ@jetpack
│    │  ├──502.54 MB (47.97%) -- js-non-window/zones
│    │  │  ├──485.42 MB (46.34%) -- zone(0x122171000)/compartment([Expanded Principal], [anonymous sandbox] (from: resource://gre/modules/addons/XPIProvider.jsm -> jar:file:///Users/erik/Library/Application%20Support/Firefox/Profiles/kb1qxu2j.default/extensions/jid1-MnnxcxisBPnSXQ@jetpack.xpi!/bootstrap.js -> resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/loader/sandbox.js:32))
│    │  │  │  ├──480.24 MB (45.84%) -- classes
│    │  │  │  │  ├──455.52 MB (43.48%) -- class(Object)
│    │  │  │  │  │  ├──233.83 MB (22.32%) -- objects
│    │  │  │  │  │  │  ├──124.39 MB (11.87%) ── malloc-heap/slots
│    │  │  │  │  │  │  └──109.44 MB (10.45%) ── gc-heap
│    │  │  │  │  │  └──221.70 MB (21.16%) -- shapes
│    │  │  │  │  │     ├──165.87 MB (15.83%) -- gc-heap
│    │  │  │  │  │     │  ├──134.09 MB (12.80%) ── dict
│    │  │  │  │  │     │  ├───19.43 MB (01.85%) ── base
│    │  │  │  │  │     │  └───12.35 MB (01.18%) ── tree
│    │  │  │  │  │     └───55.83 MB (05.33%) -- malloc-heap
│    │  │  │  │  │         ├──51.63 MB (04.93%) ── dict-tables
│    │  │  │  │  │         └───4.20 MB (00.40%) ++ (2 tiny)
│    │  │  │  │  ├───18.02 MB (01.72%) -- class(Array)
│    │  │  │  │  │   ├──18.02 MB (01.72%) -- objects
│    │  │  │  │  │   │  ├──18.02 MB (01.72%) ── gc-heap
│    │  │  │  │  │   │  └───0.00 MB (00.00%) ++ malloc-heap
│    │  │  │  │  │   └───0.00 MB (00.00%) ++ shapes
│    │  │  │  │  └────6.69 MB (00.64%) ++ (3 tiny)
│    │  │  │  └────5.18 MB (00.49%) ++ (4 tiny)
│    │  │  ├───15.67 MB (01.50%) ++ zone(0x100490000)
│    │  │  └────1.45 MB (00.14%) ++ (9 tiny)
│    │  └────1.67 MB (00.16%) ++ window-objects/top(resource://jid1-mnnxcxisbpnsxq-at-jetpack/privacybadgerfirefox/data/popup.html, id=19)

**Comment by cooperq** [comment]
I've been unable to reproduce this so far. Is it still a problem in the latest version of firefox?

**Comment by arrarrlol** [comment]
It uses about 1.6 MB now.

I suspect the issue was within Firefox's about:memory tool. The reason I think it was a Firefox bug is because after I disabled Privacy Badger, another one of my addons was reportedly using 100MB RAM. The rest of the addons were tiny. After I disabled that 100MB addon, another one of the addons, which was previous tiny, was reported to use ~80MB RAM. I disabled that too and guess what, another previously tiny addon showed as using lots of RAM. I do not think the addons were actually using that much RAM, it was most likely a bug in about:memory.

Using Firefox 38.0.1 and do not have that issue at the moment.
